miprometheus/problems/question_context_to_class/babiqa_dataset_single_question.py

```diff
@@ -0,0 +1,514 @@
+                        zfile.extractall(path)
+
+                # tarfile cannot handle bare .gz files
+                elif ext == '.tgz' or ext == '.gz' and ext_inner == '.tar':
+                    with tarfile.open(zpath, 'r:gz') as tar:
+                        dirs = [member for member in tar.getmembers()]
+                        tar.extractall(path=path, members=dirs)
+
+                #in case it is a gz file
+                elif ext == '.gz':
+                    with gzip.open(zpath, 'rb') as gz:
+                        with open(zroot, 'wb') as uncompressed:
+                            shutil.copyfileobj(gz, uncompressed)
+
+        #Return path to extracted dataset
+        return os.path.join(path, self.dirname)
+
+
+    def parse(self, file_data, add_punctuation):
+
+        """This method is parsing the file
+               :param file_data : data file to  be parsed
+               :param add_punctuation : boolean to decide wether we add punctuation 
+               :return: data : Parsed data
+           
+        """
+        #make empty lists
+        data, story,  story2 = [],[],[]
+        i = 0
+
+        #open file
+        with open(file_data, 'r') as f:
+            for line in f:
+                # print(line)
+                tid, text = line.rstrip('\n').split(' ', 1)
+                if tid == '1':
+                    story = []
+                    story2 = []
+                    answers = []
+                # don't delete period
+                if text.endswith('.'):
+                    for a in text[:-1].split():
+                        assert not isinstance(a, list)
+                        story.append(a)
+                    if add_punctuation:
+                        story.append('.')
+                else:
+                    # remove any leading or trailing whitespace after splitting
+                    query, answer, supporting = (x.strip() for x in text.split('\t'))
+
+                    for a in query[:-1].split(' '):
+                        story2.append(a)
+                    if add_punctuation:
+                        story2.append('?')
+                    for a in answer.split(','):
+                        answers.append(a)
+                        story2.extend(['_'])
+
+                    story_f = list(story)
+                    story_f.extend(story2)
+                    if story_f:
+                        data.append((story_f, answers))
+
+                        #Set answers and story back to empty lists
+                        answers = []
+                        story2 = []
+
+        return data
+
+
+
+if __name__ == "__main__":
+
+    """Unitest that generates a batch and displays a sample """
+
+    babi_tasks = list(range(1, 21))
+
+    params = {'directory': '/', 'tasks': babi_tasks,'data_type': 'train', 'batch_size': 10,'embedding_type' :'glove.6B.100d', 'ten_thousand_examples': True, 'one_hot_embedding': True, 'truncation_length':50 }
```

Comment:
Generally we are standardizing parameter names (see #73 ).
In case of data the name of variable should be "data_dir" with default value set to "~/data/babi/"

Please change that according to convention used e.g. in MNIST in __init__:
https://github.com/IBM/mi-prometheus/blob/68ad04da795c2d5e8ecb86144cb459a28ac1aa71/miprometheus/problems/image_to_class/mnist.py#L91

and then when "simulating" config file:

https://github.com/IBM/mi-prometheus/blob/68ad04da795c2d5e8ecb86144cb459a28ac1aa71/miprometheus/problems/image_to_class/mnist.py#L205

Reply:
got it